tuf/api/metadata.py

```diff
@@ -1123,6 +1123,73 @@ def to_dict(self) -> Dict[str, Any]:
             **self.unrecognized_fields,
         }
 
+    @classmethod
+    def from_file(
+        cls, filepath: str, hash_algorithms: List[str]
+    ) -> "TargetFile":
+        """Creates TargetFile object from a file.
+
+        Attributes:
+            filepath: The path to the file to create the TaretFile object from.
+            hash_algorithms: The hash algorithms to calculate the hashes with.
+                If empty, the securesystemslib default hash algorithm is used.
+
+        Raises:
+            FileNotFoundError: The file doesn't exist.
+        """
+        with open(filepath, "rb") as file:
+            return cls.from_data(file, hash_algorithms)
+
+    @classmethod
+    def from_data(
+        cls, data: Union[bytes, BinaryIO], hash_algorithms: List[str]
+    ) -> "TargetFile":
```

Comment:
If you want to support a default hash algorithms value (as I said it isn't strictly necessary), it should work like Python default values elsewhere:

`hash_algorithms: Optional[List[str]] = None`

(and then the test call for that becomes `TargetFile.from_data(data)`)

Reply:
Thanks, @jku. Addressed.